homeassistant/helpers/config_validation.py

```diff
@@ -65,9 +68,36 @@ def boolean(value: Any) -> bool:
     return bool(value)
 
 
-def isfile(value):
+def isfile(value: Any) -> str:
     """Validate that the value is an existing file."""
-    return vol.IsFile('not a file')(value)
+    if value is None:
+        raise vol.Invalid('None is not file')
+    return vol.IsFile('not a file')(str(value))
+
+
+def file(value: Any) -> str:
+    """Validate if file is reade/writeable and could be create (platform)."""
+    if value is None:
+        raise vol.Invalid('None is not file')
+    file_in = str(value)
+    (path, name) = os.path.split(file_in)
+
+    # if use HA config directory
+    if not path:
+        return file_in
+    if not name or os.path.isdir(file_in):
+        raise vol.Invalid('None is not file')
+    # file exists, control access
+    if os.path.isfile(file_in) and os.access(file_in, os.R_OK | os.W_OK):
```

Comment:
file is for check if writeable and can be creat it. I can make a new fuction like readfile. They should be cleaner as the file function.